diff --git a/libsakura/CMakeLists.txt b/libsakura/CMakeLists.txt
index 3411ea8..daa26be 100644
--- a/libsakura/CMakeLists.txt
+++ b/libsakura/CMakeLists.txt
@@ -70,7 +70,7 @@ if(APPLE)
   endif("${isSystemDir}" STREQUAL "-1")
 endif()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -DDISABLE_ALIGNAS")
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     if(USE_LIBCXX)
     	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
diff --git a/libsakura/src/libsakura/localdef.h b/libsakura/src/libsakura/localdef.h
index 098f088..4268c5d 100644
--- a/libsakura/src/libsakura/localdef.h
+++ b/libsakura/src/libsakura/localdef.h
@@ -179,7 +179,7 @@ private:
   #define __has_builtin(x) 0  // Compatibility with non-clang compilers.
 #endif
 
-#if defined(__GNUG__) && (!defined(__clang__) || __has_builtin(__builtin_assume_aligned))
+#if defined(__GNUG__) && !defined(__clang__) && __has_builtin(__builtin_assume_aligned)
 /**
  * @~japanese
  * @brief @a ptr がsakuraのアライメント要件を満たしていると見なし、そのアドレスを返す。
diff --git a/libsakura/src/convolution.cc b/libsakura/src/convolution.cc
index 803ef12..278ebaa 100644
--- a/libsakura/src/convolution.cc
+++ b/libsakura/src/convolution.cc
@@ -38,6 +38,7 @@
 #include <climits>
 #include <fstream>
 #include <sstream>
+#include <stdexcept>
 
 #include <libsakura/sakura.h>
 #include <libsakura/localdef.h>
diff --git a/libsakura/src/gridding.cc b/libsakura/src/gridding.cc
index 9953255..6e3f68a 100644
--- a/libsakura/src/gridding.cc
+++ b/libsakura/src/gridding.cc
@@ -27,6 +27,7 @@
 #include <cassert>
 #include <cmath>
 #include <climits>
+#include <cstdint>
 
 #include "libsakura/sakura.h"
 #include "libsakura/localdef.h"
